### PR TITLE
NG firewall conversion

### PR DESCRIPTION
**Pull request by radkrawczyk**
# Pull Request summary:
Introducing ZoneBasedFirewall converter

# Checklist:
- [x] Make sure to run pre-commit before committing changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

File: catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        parcel = NgfirewallParcel.create(
+            parcel_name=in_.name,
+            parcel_description=in_.description,
+            default_action_type=in_.definition.default_action.type,
+            sequences=sequences,
+        )
+        convert_result.output = parcel
+
+    except Exception as e:
```
Comment (jpkrajewski) [inline catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py:172]: The broad `except Exception` clause could be more specific to catch expected exceptions.

File: catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    elif match_entry.field == "sourcePort":
+        return SourcePort.from_str_list(match_entry.value.split(" "))
+    elif match_entry.field == "destinationPort":
+        return DestinationPort.from_str_list(match_entry.value.split(" "))
+    elif match_entry.field == "sourceGeoLocation":
+        return SourceGeoLocation.from_geo_locations_list(cast_(match_entry.value.split(" ")))
+    elif match_entry.field == "destinationGeoLocation":
+        return DestinationGeoLocation.from_geo_locations_list(cast_(match_entry.value.split(" ")))
+    elif match_entry.field == "protocolName":
+        return ProtocolNameMatch.from_protocol_name_list(cast_(match_entry.value.split(" ")))
```
Comment (jpkrajewski) [inline catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py:105]: Consider moving `cast_(match_entry.value.split(" ")` to helper function

**Review by sbasan — APPROVED** [review]

File: catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py (hunk 1 above)
Comment (sbasan) [reply catalystwan/utils/config_migration/converters/policy/zone_based_firewall.py:172]: looks like `ValueError` was added to come construction methods

File: catalystwan/models/configuration/feature_profile/sdwan/embedded_security/ngfirewall.py [hunk 3]
```diff
@@ -716,6 +842,37 @@ class Sequence(BaseModel):
         serialization_alias="disableSequence",
     )
 
+    def add_log_action(self) -> None:
+        if LogAction in map(type, self.actions):
+            raise ValueError("LogAction is already added to NGfirewall sequence")
```
Comment (sbasan) [inline catalystwan/models/configuration/feature_profile/sdwan/embedded_security/ngfirewall.py:847]: can we move this check to `actions` validator as an assertion so `ValidationError` will be used which i believe is correct type for such exception (Also convert results are validated again after conversion and there is already a mechanism to report such errors)

**Pull request merged**